crates/bevy_pbr/src/render/pbr_functions.wgsl

```diff
@@ -223,7 +205,25 @@ fn apply_pbr_lighting(
                 && (view_bindings::point_lights.data[light_id].flags & mesh_view_types::POINT_LIGHT_FLAGS_SHADOWS_ENABLED_BIT) != 0u) {
             shadow = shadows::fetch_point_shadow(light_id, in.world_position, in.world_normal);
         }
-        let light_contrib = lighting::point_light(in.world_position.xyz, light_id, roughness, NdotV, in.N, in.V, R, F0, f_ab, diffuse_color);
+        let light_contrib = lighting::point_light(
+            in.world_position.xyz,
+            light_id,
+            roughness,
+            NdotV,
+            in.N,
+            in.V,
+            R,
+            F0,
+            f_ab,
+#ifdef STANDARD_MATERIAL_CLEARCOAT
+            clearcoat_NdotV,
+            clearcoat_N,
+            clearcoat_R,
+            clearcoat,
+            clearcoat_roughness,
+#endif  // STANDARD_MATERIAL_CLEARCOAT
+            diffuse_color
```

Comment:
maybe we could have a struct to hold all these arguments ... or most of them, so we could reuse it with a couple extra params for all the lighting calls?

Reply:
Working on this. This is a huge change :)

Reply:
This is now done. The entire `pbr_lighting.wgsl` file has been refactored from top to bottom to use structures throughout. I also changed `environment_map.wgsl` to do the same